Formula/c/clang-uml.rb

```diff
@@ -0,0 +1,123 @@
+class ClangUml < Formula
+  desc "Customizable automatic UML diagram generator for C++ based on Clang"
+  homepage "https://github.com/bkryza/clang-uml"
+  url "https://github.com/bkryza/clang-uml/archive/refs/tags/0.5.3.tar.gz"
+  sha256 "e830363ec510f14cc738c6509107b3f52bc55ececc2e27c068cadb093604e943"
+  license "Apache-2.0"
+  head "https://github.com/bkryza/clang-uml.git", branch: "master"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "pkg-config" => :build
+  depends_on "llvm"
+  depends_on "yaml-cpp"
+
+  fails_with gcc: "5"
+
+  def llvm
+    deps.map(&:to_formula)
+        .find { |f| f.name.match?(/^llvm(@\d+)?$/) }
+  end
+
+  def install
+    args = %W[
+      -DBUILD_TESTS=OFF
+      -DCLANG_UML_ENABLE_BACKTRACE=OFF
+      -DCMAKE_INSTALL_RPATH=#{rpath(source: libexec/"bin", target: llvm.opt_lib)}
+    ]
+
+    # If '.git' directory is not available during build, we need
+    # to provide the version using a CMake option
+    args.append("-DGIT_VERSION=#{version}") unless build.head?
+
+    # clang-uml needs to be linked to LLVM c++ when compiled with Clang, otherwise
+    # Trace/BPT trap is raised whenever an exception is thrown (even if it's handled properly)
+    if OS.mac?
+      cc_bin = llvm.opt_bin/"clang"
+      cxx_bin = llvm.opt_bin/"clang++"
+      args.append(%Q(-DCMAKE_EXE_LINKER_FLAGS="-L#{llvm.opt_lib}/c++ -Wl,-rpath,#{llvm.opt_lib}/c++"))
+    end
+
+    with_env(CC: cc_bin, CXX: cxx_bin) do
+      system "cmake", "-S", ".", "-B", "build", *std_cmake_args(install_prefix: libexec), *args
+    end
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    bin.write_exec_script libexec.glob("bin/*")
+
+    # Remove redundant files
+    rm_f Dir["#{share}/doc/clang-uml/LICENSE.md"]
+    rm_f Dir["#{share}/doc/clang-uml/AUTHORS.md"]
+    rm_f Dir["#{share}/doc/clang-uml/CHANGELOG.md"]
+    rm_f Dir["#{share}/doc/clang-uml/README.md"]
+
+    bash_completion.install "packaging/autocomplete/clang-uml"
+    zsh_completion.install "packaging/autocomplete/_clang-uml"
+  end
+
+  def caveats
+    on_macos do
+      <<~EOS
+        If you see errors such as `fatal: 'stddef.h' file not found`, try
+        adding `--query-driver .` to the `clang-uml` command line in order to
+        ensure that proper system headers are available to Clang.
+
+        For more information see: https://clang-uml.github.io/md_docs_2troubleshooting.html
+      EOS
+    end
+  end
+
+  test do
+    # Check if clang-uml is linked properly
+    shell_output("#{bin}/clang-uml --version 2>&1")
+    shell_output("#{bin}/clang-uml --help 2>&1")
```

Comment:
```suggestion
    system bin/"clang-uml", "--version"
    system bin/"clang-uml, "--help"
```
since you're not doing anything with the output.